### ISSUE TITLE
Source Map Line Numbers Are Wrong

### ISSUE THREAD

**Issue by machineghost** [vercel/next.js#21904]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

15.4.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Linux

**How are you deploying your application?**

I'm having the problem in a dev environment; deployment is irrelevant

**Describe the Bug**

Basically it seems bug #2285 is back.  If I get an error from *actual* line 24 of a file, the stack trace generated shows it coming from line 28 ... which is an empty line in the original source.

I've often seen this sort of thing when using the "wrong" sourcemap settings in Webpack, and I'd be more than happy to try cycling through all the various options to report back which one results in the correct line numbers ... only I have no idea how to tell Next.js what source map settings to give to Webpack.

**Expected Behavior**

When I get an error in dev, the line number should correspond to the line number in the file (or at least, after the browser translates that line through the source map).

**To Reproduce**

I'm observing this issue in the middle of a larger app, which I can't submit.  However, the one interesting part is that it's coming from a local shared library.

In other words, I have project Foo, and also project Bar in the same folder.  Bar has the code for the function (with the incorrect stack trace), but said function is called by Foo.  Foo is able to do this because I did `npm i ../Bar`.

I realize this is a specific case that others might not have, and again I'm happy to debug on my end to try and get find the correct `devtool` (https://webpack.js.org/configuration/devtool/) to resolve it ... I just need to know how I can use different `devtool` settings with Webpack when using Next.js

**Comment by machineghost** [comment]
Nevermind ... stopping and restarting my devsite seems to have fixed things.  It'd still be nice to learn how to try other webpack settings (of any sort, not just devtool) with Next.js, if anyone wants to share, but I'll close this issue since even I can't reproduce it now.

**Comment by machineghost** [comment]
Re-opening: this is an odd one.  When I go from my index page to the page causing the error, I get a normal stack trace with correct line numbers.  But when I go straight to that problem page (eg. by reloading while on it), I get the incorrect stack trace.

This doesn't make a lot of sense to me, and I still don't have clear reproduction steps ... but I'm re-opening in case this makes more sense to someone on the Next team.

**Comment by jasonnet** [comment]
This can also be reproduced using the Next.js tutorial or just create-next-app.   Simply add a  "throw new Error()" to ./pages/api/hello.js   .  Then refresh that page and watch the console.    I've confirmed this on Mac and Linux.  And on Next.js 10.x and 11.0.     For some combinations of device and configuration, you might need to add an  import at the top of hello.js to reproduce the line number mismatch.

I'm also seeing that if one adds code there to grab a stack trace there and snoop it, it's not the same trace as on the console, but neither of them are correct.

I assume this is related to webpack and babel.    I've tried a variety of guidance provided on the web and so far nothing fixes this.

**Comment by samuela** [comment]
Still an issue as of `next@12.0.2` and node.js v15.14.0.

**Comment by samuela** [comment]
Other reports of this issue:
- https://www.reddit.com/r/nextjs/comments/l48pp3/are_stack_traces_on_api_routes_broken/
- https://github.com/vercel/next.js/discussions/15920
- https://stackoverflow.com/questions/68011227/how-to-achieve-proper-server-side-stack-trace-line-numbers-in-next-js

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
